fvm/src/call_manager/default.rs

```diff
@@ -427,9 +432,35 @@ where
             },
         };
 
+        // TODO should we deduct the method invocation gas from the supplied gas_limit?
+        //  Assuming yes -- this happens inside send_resolved.
+
+        // If a specific gas limit has been requested, create a child GasTracker and use that
+        // one hereon.
+        let prev_gas_tracker = match gas_limit {
+            Some(gas_limit) if gas_limit < self.gas_tracker.gas_available() => {
+                let gas_used = self.gas_tracker.gas_used();
+                let tracing = self.machine.context().tracing;
+                let new = GasTracker::new(gas_limit, gas_used, tracing);
+                Some(mem::replace(&mut self.gas_tracker, new))
+            }
+            _ => None,
+        };
+
         // Do the actual send.
+        let ret = self.send_resolved::<K>(from, to, method, params, value);
+
+        // Restore the original gas tracker and absorb the child's gas usage and traces into it.
+        if let Some(prev) = prev_gas_tracker {
+            let other = mem::replace(&mut self.gas_tracker, prev);
+            // This is capable of raising an OutOfGas, but it is redundant since send_resolved
+            // would've already raised it, so we ignore it here. We could check and assert that's
+            // true, but send_resolved could also error _fatally_ and mask the OutOfGas, so it's
+            // not safe to do so.
+            let _ = self.gas_tracker.absorb(&other);
```

Comment:
If we made the limit updatable, we wouldn't have to do this.

Reply:
Could do, although that kind of mutation moves the accounting outside the component? Seems cleaner this way (but also less performant).

Reply:
I think a `lower_limit` and `raise_limit` operation makes sense.